client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java

```diff
@@ -163,6 +163,11 @@
      */
     private int consumeThreadMax = 20;
 
+    /**
+     * Queue size of consumption request;
+     */
+    private int consumeQueueSize = 10000;
```

Comment:
If we set up a bounded queue, one issue that needs to be considered is how to choose the rejection strategy when the queue is full, because this may cause message loss.

Reply:
Thanks for reminding.

The original rejection strategy is good. If the caller catch an RejectedExecutionException, it try to submit again after a delay. and the delay increases exponentially until it reaches a maximum.
This PR has implementation and unit tests that don't cause messages to be lost.